File: src/main/java/io/github/nikpivkin/lifetime/impl/PluginServiceImpl.java
package io.github.nikpivkin.lifetime.impl;

import io.github.nikpivkin.lifetime.PluginConfig;
import io.github.nikpivkin.lifetime.PluginService;
import io.github.nikpivkin.lifetime.events.GodModeChangedEvent;
import io.github.nikpivkin.lifetime.events.PluginReloadEvent;
import io.github.nikpivkin.lifetime.events.PluginStateChangedEvent;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class PluginServiceImpl implements PluginService {

  private final PluginConfig config;

  public PluginServiceImpl(PluginConfig config) {
    this.config = config;
  }

  @Override
  public void enable() {
    config.setEnabled(true);
    Bukkit.getPluginManager().callEvent(new PluginStateChangedEvent.Enabled());
  }

  @Override
  public void disable() {
    config.setEnabled(false);
    Bukkit.getPluginManager().callEvent(new PluginStateChangedEvent.Disabled());
  }

  @Override
  public void excludeUser(Player player) {
    config.excludeUser(player.getName());
    Bukkit.getPluginManager().callEvent(new GodModeChangedEvent.Disabled(player));
  }

  @Override
  public void includeUser(Player player) {
    config.includeUser(player.getName());
    Bukkit.getPluginManager().callEvent(new GodModeChangedEvent.Enabled(player));
  }

  @Override
  public void reload() {
    config.reload();
    Bukkit.getPluginManager().callEvent(new PluginReloadEvent());
  }
}

File: src/main/java/io/github/nikpivkin/lifetime/impl/PluginServiceImpl.kt
package io.github.nikpivkin.lifetime.impl

import io.github.nikpivkin.lifetime.PluginConfig
import io.github.nikpivkin.lifetime.PluginService
import io.github.nikpivkin.lifetime.event.GodModeChangedEvent
import io.github.nikpivkin.lifetime.event.PluginReloadEvent
import io.github.nikpivkin.lifetime.event.PluginStateChangedEvent
import org.bukkit.Bukkit
import org.bukkit.entity.Player

public class PluginServiceImpl(
    private val config: PluginConfig
) : PluginService {

    override fun enable() {
        config.setEnabled(true)
        Bukkit.getPluginManager().callEvent(PluginStateChangedEvent.Enabled())
    }

    override fun disable() {
        config.setEnabled(false)
        Bukkit.getPluginManager().callEvent(PluginStateChangedEvent.Disabled())
    }

    override fun excludeUser(player: Player) {
        config.excludeUser(player.name)
        Bukkit.getPluginManager().callEvent(GodModeChangedEvent.Disabled(player))
    }

    override fun includeUser(player: Player) {
        config.includeUser(player.name)
        Bukkit.getPluginManager().callEvent(GodModeChangedEvent.Enabled(player))
    }

    override fun reload() {
        config.reload()
        Bukkit.getPluginManager().callEvent(PluginReloadEvent())
    }
}
